### PR TITLE
Added index to Segment / SegmentReceipt

### PR DESCRIPTION
**Pull request by mothran**
Useful for reconstructing a SessionReceipt from a unordered list of SegmentReceipts.

### CODE REVIEW

File: risc0/zkvm/src/exec/mod.rs [hunk 1]
```diff
@@ -183,7 +184,9 @@ impl<'a> Executor<'a> {
                         syscalls,
                         exit_code,
                         log2_ceil(total_cycles.next_power_of_two()),
+                        idx,
```
Comment (flaub) [inline risc0/zkvm/src/exec/mod.rs:187]:
```suggestion
                        segments.len(),
```

File: risc0/zkvm/src/receipt.rs [hunk 2]
```diff
@@ -134,6 +134,9 @@ pub struct SegmentReceipt {
     /// [SegmentReceipt::verify]) to cryptographically prove that this Segment
     /// was faithfully executed.
     pub seal: Vec<u32>,
+
+    /// Segment index within the [SessionReceipt]
+    pub index: u64,
```
Comment (flaub) [inline risc0/zkvm/src/receipt.rs:139]: 4 billion segments seems like a lot :)

**Review by flaub — APPROVED** [review]

File: risc0/zkvm/src/receipt.rs (hunk 2 above)
Comment (mothran) [reply risc0/zkvm/src/receipt.rs]: I like to be optimistic hahaha, but u32 is more portable. Fixed in: c041649754ba2478d005ad8b77cd01fe1c8102d5

File: risc0/zkvm/src/exec/mod.rs (hunk 1 above)
Comment (mothran) [reply risc0/zkvm/src/exec/mod.rs]: Fixed in: c041649754ba2478d005ad8b77cd01fe1c8102d5 with additional removal of the idx var.

**Pull request merged**